release-calendar/src/client/api-service.ts

```diff
@@ -14,29 +14,35 @@
  * limitations under the License.
  */
 
-import {
-  ApiServiceInterface,
-  Promotion as PromotionEntity,
-  Release as ReleaseEntity,
-} from '../types';
-import {CurrentReleases, Release} from './models/view-models';
+import {ApiServiceInterface, Promotion} from '../types';
+import {CurrentReleases, EventInput} from './models/view-models';
 import fetch from 'node-fetch';
 const SERVER_URL = `http://localhost:3000`;
 
 export class ApiService implements ApiServiceInterface {
-  private getReleaseRequest(url: string): Promise<ReleaseEntity[]> {
+  private getPromotionRequest(url: string): Promise<Promotion[]> {
     return fetch(url).then((result) => result.json());
   }
-
-  private getPromotionRequest(url: string): Promise<PromotionEntity[]> {
+  private getPromotionRequests(url: string): Promise<Promotion[][]> {
     return fetch(url).then((result) => result.json());
   }
 
-  async getReleases(): Promise<Release[]> {
-    const releases = await this.getReleaseRequest(SERVER_URL);
-    return releases.map((release: ReleaseEntity) => {
-      return new Release(release);
+  async getReleases(): Promise<EventInput[]> {
+    const promotions = await this.getPromotionRequests(SERVER_URL);
+    const collect: EventInput[] = [];
+    promotions.forEach((promotionsInChannel: Promotion[]) => {
+      const endDate: Date = new Date();
```

Comment:
You can move this line outside of the `forEach` callback function - the way it is now it calls `new Date()` for every promotion

Reply:
Just to clarify, endDate needs to be set to a new Date (new Date is automatically set to the time its called - think current) for each row because the promotion that ends up getting set first is the most recent (I queried the promotions in descending order by date). That promotion remains the release in that channel at the time the viewer is seeing the calendar.

Reply:
I figure out that bit - but you can instantiate the constant `endDate` once, before the `forEach` line, and then use it inside the `forEach` block